idaes/models/unit_models/tests/test_gibbs.py

```diff
@@ -1013,583 +462,23 @@ def test_initialize_duty(self, methane):
-                "Vap", "O2"
-            ]
-        ] = 348491.5863157065
-        methane.scaling_factor[
-            methane.fs.unit.control_volume.properties_out[0.0].pressure
-        ] = 9.869232667160129e-06
-        methane.scaling_factor[
-            methane.fs.unit.control_volume.properties_out[0.0].temperature
-        ] = 0.0004275929186958249
-        methane.scaling_factor[
-            methane.fs.unit.gibbs_minimization[0.0, "Vap", "CH4"]
-        ] = 8.074436602893497e-07
-        methane.scaling_factor[
-            methane.fs.unit.gibbs_minimization[0.0, "Vap", "CO2"]
-        ] = 7.961846197368761e-07
-        methane.scaling_factor[methane.fs.unit.gibbs_minimization[0.0, "Vap", "CO"]] = (
-            1.306639729794169e-06
-        )
-        methane.scaling_factor[
-            methane.fs.unit.gibbs_minimization[0.0, "Vap", "H2O"]
-        ] = 1.1245365150847992e-06
-        methane.scaling_factor[methane.fs.unit.gibbs_minimization[0.0, "Vap", "H2"]] = (
-            1.9650913378522065e-06
-        )
-        methane.scaling_factor[methane.fs.unit.gibbs_minimization[0.0, "Vap", "N2"]] = (
-            1.5268799321929755e-06
-        )
-        methane.scaling_factor[
-            methane.fs.unit.gibbs_minimization[0.0, "Vap", "NH3"]
-        ] = 1.0314993300171925e-06
-        methane.scaling_factor[methane.fs.unit.gibbs_minimization[0.0, "Vap", "O2"]] = (
-            9.653850829168712e-07
-        )
-        methane.scaling_factor[methane.fs.unit.lagrange_mult[0.0, "C"]] = (
-            3.915957066473961e-06
-        )
-        methane.scaling_factor[methane.fs.unit.lagrange_mult[0.0, "H"]] = (
-            5.55811764258496e-06
-        )
-        methane.scaling_factor[methane.fs.unit.lagrange_mult[0.0, "N"]] = (
-            4.318668616445236e-06
-        )
-        methane.scaling_factor[methane.fs.unit.lagrange_mult[0.0, "O"]] = (
-            2.730521354347429e-06
-        )
+        scaler = AutoScaler(overwrite=True)
+        scaler.variables_by_magnitude(methane)
+        scaler.constraints_by_jacobian_norm(methane)
 
-        scaling = TransformationFactory("core.scale_model")
-        sm = scaling.create_using(methane, rename=False)
+        results = solver.solve(methane, writer_config={"scale_model": True})
 
-        results = solver.solve(sm)
+        # Check for optimal solution
+        assert_optimal_termination(results)
 
-        scaling.propagate_solution(sm, methane)
+    @pytest.mark.solver
+    @pytest.mark.skipif(solver is None, reason="Solver not available")
+    @pytest.mark.integration
+    def test_verify_scaling_duty(self, methane):
+        assert jacobian_cond(methane, scaled=True) == pytest.approx(9191, abs=1)
 
-        # Check for optimal solution
-        assert check_optimal_termination(results)
+        assert len(extreme_jacobian_rows(methane, scaled=True)) == 0
+        assert len(extreme_jacobian_rows(methane, scaled=True)) == 0
```

Comment:
Actually, adding to my other suggestion on thinking about a reporting method for `Autoscaler`, these would also be good to show in report output (e.g., condition number, extreme jac rows/columns or at least the number of each). 

I realize that the DiagnosticsToolbox already has the capability to report on some of these things, but just thinking it'd be nice to be able to apply scaling via AutoScaler and subsequently check what you really did to model scaling with a report method.

Reply:
That is what the Diagnostics tools are for - putting those methods here would just be duplication of code (and the diagnostics also tell you a lot more). The eventual documentation will highlight that you really need to use diagnostics, scaling and initialization tools together to get the best results.

Reply:
Will the diagnostics toolbox reference these new tools?

Reply:
@agarciadiego I am not sure there - we probably need to think about how and where we draw the line between the two. Up until now, the Diagnostics Toolbox has been about finding the issues, and hasn't said much about how to fix them (as that is not so easy to explain concisely). We can definitely mention it in the docs, and maybe we could have the `display` methods mention tools to help fix issues (where appropriate).